### DISCUSSION TITLE
Creating cable labels using API and pynetbox

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#11476] [General]
Hi

I am just trying to make a little script, that takes the interfaces of a device that has cables defined and print the termination-points. I need this information to piece together the label, that we are going to print out and put on the cable itself.
I have gotten to point of fetching the cable from Netbox, but I cannot figure out how to print the tuple information about just the a and b termination points.

`device_name = TESTSRV`
`nb_device = nb.dcim.devices.get(name=device_name)`
`if not nb_device:`
`    print ("Device not found in netbox")`
`    exit()`
`nb_device_name = str(nb_device.name)`
`nb_device_interfaces = nb.dcim.interfaces.filter(device=nb_device_name)`
`for nb_interface in nb_device_interfaces:`
`      nb_interface_cable = nb_interface.cable`
`      if nb_interface_cable == None:`
`            continue`
`      else:`
`            print("interface_cable id",nb_interface_cable.id)`
`            nb_cable = nb.dcim.cables.get(nb_interface_cable.id)`
`            for output in nb_cable:`
`               print (output)`
`            print("interface cable",nb_cable)`

Output:

> device_name TESTSRV
> interface_cable id 2118
> ('id', 2118)
> ('url', 'https://netbox.test/api/dcim/cables/2118/')
> ('display', '#2118')
> ('type', '')
> **('a_terminations',** [{'object_type': 'circuits.circuittermination', 'object_id': 165, 'object': {'id': 165, 'url': 'https://netbox.test/api/circuits/circuit-terminations/165/', 'display': 'Termination Z: TEST-SITE', 'circuit': {'id': 102, 'url': 'https://netbox.test/api/circuits/circuits/102/', 'display': 'TEST-CIRCUIT-1', 'cid': 'TEST-CIRCUIT-1'}, 'term_side': 'Z', 'cable': 2118, '_occupied': True}}])
> **('b_terminations'**, [{'object_type': 'dcim.interface', 'object_id': 58965, 'object': {'id': 58965, 'url': 'https://netbox.test/api/dcim/interfaces/58965/', 'display': 'et-0/0/0', 'device': {'id': 1517, 'url': 'https://netbox.test/api/dcim/devices/1517/', 'display': 'TESTSRV', 'name': 'TESTSRV'}, 'name': 'et-0/0/0', 'cable': 2118, '_occupied': True}}])
> ('status', {'value': 'connected', 'label': 'Connected'})
> ('tenant', {'id': 764, 'url': 'https://netbox.test/api/tenancy/tenants/764/', 'display': 'TEST-TENANT', 'name': 'TEST-TENANT', 'slug': 'test-tenant'})
> ('label', '')
> ('color', '')
> ('length', None)
> ('length_unit', None)
> ('tags', [])
> ('custom_fields', {})
> ('created', '2023-01-11T14:01:41.398805+01:00')
> ('last_updated', '2023-01-11T14:09:17.594893+01:00')
> interface cable Cable #2118

If I try something like printing nb_cable.a_terminations I just get empty output.
I need the a and b terminations a list or dict, so I can pick and choose from the contents. Any ideas?

**Comment by candlerb** [comment]
You need to upgrade pynetbox.

I can reproduce your problem here with pynetbox 6.1.2:

```
>>> c = nb.dcim.cables.get(66)
>>> c
Cable #66
>>> c.a_terminations
[]
```

But after upgrading to pynetbox 7.0.0:

```
>>> c = nb.dcim.cables.get(66)
>>> c
enp3s0 <> ether3
>>> c.a_terminations
[enp3s0]
```

**Reply by nomad-cyanide** [reply]
Thanks. That did the trick.
